snmp/datadog_checks/snmp/snmp.py

```diff
@@ -122,6 +122,8 @@ def _load_profiles(self):
 
             try:
                 recursively_expand_base_profiles(definition)
+            except RecursionError as exc:
```

Comment:
Handling recursion "gracefully" is needed to handle `palo-alto.yaml`, this will help skip the profile with a warning instead of failing the whole integration completely.